Rubberduck.VBEEditor/ComManagement/TypeLibs/VBETypeLibsIterator.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Rubberduck.VBEditor.ComManagement.TypeLibs
+{
+    /// <summary>
+    /// An internal representation of the ITypeLib object hosted by the VBE.
+    /// Also provides Prev/Next pointers, exposing a double linked list of all loaded project ITypeLibs
+    /// </summary>
+    [StructLayout(LayoutKind.Sequential)]
+    struct VBETypeLibObj
+    {
+        IntPtr _vTable1;     // ITypeLib vtable
+        IntPtr _vTable2;
+        IntPtr _vTable3;
+        public IntPtr Prev;
+        public IntPtr Next;
+    }
+
+    /// <summary>
+    /// An enumerable class for iterating over the double linked list of ITypeLibs provided by the VBE 
+    /// </summary>
+    public sealed class VBETypeLibsIterator : IEnumerable<TypeLibWrapper>, IEnumerator<TypeLibWrapper>
+    {
+        private IntPtr _currentTypeLibPtr;
+        private VBETypeLibObj _currentTypeLibStruct;
+        private bool _isStart;
+
+        public VBETypeLibsIterator(IntPtr typeLibPtr)
+        {
+            _currentTypeLibPtr = typeLibPtr;
+            _currentTypeLibStruct = StructHelper.ReadStructureSafe<VBETypeLibObj>(_currentTypeLibPtr);
+            Reset();
+        }
+
+        public void Dispose()
+        {
+            // nothing to do here, we don't own anything that needs releasing
+        }
+
+        IEnumerator IEnumerable.GetEnumerator() => this;
+        public IEnumerator<TypeLibWrapper> GetEnumerator() => this;
+
+        TypeLibWrapper IEnumerator<TypeLibWrapper>.Current => new TypeLibWrapper(_currentTypeLibPtr, makeCopyOfReference: true);
+        object IEnumerator.Current => new TypeLibWrapper(_currentTypeLibPtr, makeCopyOfReference: true);
```

Comment:
The parameters here were changed from `false` to `true`. Reverting that problems avoids the memory corruption but still introduces an AV.

Reply:
In the end, the current implementation is correct and does not need revising.